app/src/main/java/org/openimis/imispolicies/ToRestApi.java

```diff
@@ -12,137 +14,116 @@
+        httpGet.setHeader("Content-Type", "application/json");
+        httpGet.setHeader("accept", "application/json");
+        httpGet.setHeader("api-version", apiVersion);
+        if (addToken) {
+            httpGet.setHeader("Authorization", "bearer " + token.getTokenText().trim());
         }
 
-        HttpResponse response = null;
         try {
-            response = httpClient.execute(httpPost);
+            HttpResponse response = httpClient.execute(httpGet);
+            if (addToken && response != null && response.getStatusLine().getStatusCode() == HttpURLConnection.HTTP_UNAUTHORIZED) {
+                token.clearToken();
+            }
+            int responseCode = response.getStatusLine().getStatusCode();
+            Log.i("HTTP_GET", uri + functionName + " - " + responseCode);
+            return response;
         } catch (IOException e) {
             e.printStackTrace();
+            return null;
         }
-
-        return getContent(response);
     }
 
-    // Post with Token JWT, returned response
-    public HttpResponse postToRestApiToken(final JSONObject object, final String functionName) {
+    public HttpResponse postToRestApi(JSONObject object, String functionName, boolean addToken) {
         HttpClient httpClient = new DefaultHttpClient();
-        HttpPost httpPost = new HttpPost(uri+functionName);
-        try {
-            StringEntity postingString = new StringEntity(object.toString());
-            httpPost.setEntity(postingString);
-            httpPost.setHeader("Content-type", "application/json");
-            httpPost.setHeader("Authorization", "bearer "+tokenl.getTokenText());
-        } catch (UnsupportedEncodingException e) {
-            e.printStackTrace();
+
+        HttpPost httpPost = new HttpPost(uri + functionName);
+        httpPost.setHeader("Content-type", "application/json");
+        httpPost.setHeader("accept", "application/json");
+        httpPost.setHeader("api-version", apiVersion);
+        if (addToken) {
+            httpPost.setHeader("Authorization", "bearer " + token.getTokenText().trim());
         }
 
-        HttpResponse response = null;
         try {
-            response = httpClient.execute(httpPost);
-            if (response != null && response.getStatusLine().getStatusCode() == HttpURLConnection.HTTP_UNAUTHORIZED) {
-                tokenl.clearToken();
+            if (object != null) {
+                StringEntity postingString = new StringEntity(object.toString());
+                httpPost.setEntity(postingString);
             }
+            HttpResponse response = httpClient.execute(httpPost);
+            if (addToken && response != null && response.getStatusLine().getStatusCode() == HttpURLConnection.HTTP_UNAUTHORIZED) {
+                token.clearToken();
+            }
+            int responseCode = response.getStatusLine().getStatusCode();
+            Log.i("HTTP_POST", uri + functionName + " - " + responseCode);
+            return response;
```

Comment:
I see similarity in this code with the method above. Please try to find common code and move to the separate method